frontend/server/src/DAO/Problems.php

```diff
@@ -482,6 +482,12 @@ final public static function byIdentityType(
                 public: true,
                 showUserTags: $row['allow_user_add_tags']
             );
+            $problem['difficulty_histogram'] = is_null(
+                $problem['difficulty_histogram']
+            ) ? [] : $problem['difficulty_histogram'];
```

Comment:
Hay forma de convertir esta cadena a un array:
```suggestion
            ) ? [] : json_decode($problem['difficulty_histogram']);
```
?

De esta forma será más fácil maipular los datos desde el cliente que se utilice.

Mismo caso para `difficulty_histogram`

Reply:
no estoy seguro, la anotación en la linea 85 pone a ambos histogramas como un array de enteros, me parece que sería más complicado intentar esa conversión, en ese caso no se si sería posible hacer lo opuesto y cambiar la anotacion para que estos retornen directamente la cadena json

Reply:
Ok, entiendo el punto, pero en ese caso le delegaríamos la responsabilidad al cliente de convertirlo al tipo correcto cuando lo conveniente es que regresaramos el campo de la forma en la que este lo pueda utilizar sin tener que manipularlo. 

Así que vamos a resolverlo paso a paso:

- En el campo de `difficulty_histogram` y `quality_histogram` se obtiene `null` o una cadena de texto con un formato en json que contiene 5 valores numericos, por ejemplo: `[0, 0, 0, 0, 2]`. Hasta aquí es una cadena de texto, tal como se describe en la anotación de la línea 451.
- Si el valor que obtenemos es `null` le podemos asignar un array vacío para que coincida con la declaración en la línea 85
- En caso contrario podemos decodificar el json que se obtiene de la base de datos. Algo muy importante es que `PSALM` no conoce la estructura que se obtendrá del json, así que podemos definir el tipo manualmente, en este caso sería una lista de enteros.
- Por último podemos agregar una validación extra que nos arroje un error si el número de elementos que se obtiene en el json es distinto a 5. Este número siempre deberá ser 5, ya que estos son los niveles que tenemos definidos tanto de dificultad como de calidad de un problema cuando un usuario lo califica.

Entonces el código podría cambiar de la siguiente forma:

```php
            $difficultyHistogram = [];
            if (!is_null($row['difficulty_histogram'])) {
                /** @var list<int> */
                $difficultyHistogram = json_decode($row['difficulty_histogram']);
                if (count($difficultyHistogram) !== 5) {
                    throw new \OmegaUp\Exceptions\InvalidParameterException(
                        'parameterInvalid',
                        'difficulty_histogram'
                    );
                }
            }

            $problem['difficulty_histogram'] = $difficultyHistogram;
``` 

Lo mismo aplicaría para el campo `quality_histogram`.

De esta forma `PSALM` no tendría por que quejarse

Reply:
Entendido, lo implementare de esa forma

Reply:
ok, intente aplicar tu sugerencia pero me produce un error de php en la función json_decode, no he ubicado la linea exacta en que ocurre pero parece ser que `difficulty_histogram `y `quality_histogram` ya fueron convertidos a arreglos de enteros para cuando llegan a la función json_decode

el error en cuestion:
 ```
ERROR: InvalidArgument - frontend/server/src/DAO/Problems.php:489:21 - Argument 1 of json_decode expects string, but list<int> provided (see https://psalm.dev/004)
                    $problem['difficulty_histogram']
```